lib/rubocop/cop/style/redundant_arguments.rb

```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # This cop checks for redundant arguments to methods.
+      # Method names & arguments can be configured like:
+      #
+      # RedundantArguments:
+      # - Method: "join"
+      #   Argument: ""
+      # - Method: "split"
+      #   Argument: " "
+      # - Method: "foo"
+      #   Argument: 2
+      #
+      # @example
+      #   # bad
+      #
+      #   array.join('')
+      #   [1, 2, 3].join("")
+      #   string.split(" ")
+      #   "first\nsecond".split(" ")
+      #   A.foo(2)
+      #
+      #   # good
+      #   array.join
+      #   [1, 2, 3].join
+      #   string.split
+      #   "first second".split
+      #   A.foo
+      class RedundantArguments < Cop
+        MSG = 'Use `%<without_argument>s` instead of `%<with_argument>s`.'
+
+        def on_send(node)
+          config = config_for(node.method_name)
+          return unless config && argument_match?(config, node)
+
+          message = format(MSG, without_argument: node.method_name,
+                                with_argument: node.source.split('.')[1])
+          add_offense(node, message: message)
+        end
+
+        private
+
+        def config_for(method_name)
+          cop_config['RedundantArguments'].find do |c|
+            c['Method'] == method_name.to_s
+          end
+        end
+
+        def argument_match?(config, node)
+          return unless config
+
+          config['Argument'].to_s == unwrap(node.arguments[0]&.source)
+        end
+
+        # Parser wraps empty strings inside quotes
+        # this is a workaround to compare
+        def unwrap(string)
+          case string
+          when "''", '""'
+            ''
+          when '" "', "' '"
+            ' '
```

Comment:
This is a bad workaround & will fail with cases like `'    '` (string with spaces). Can anyone help me fix this?

Reply:
I guess first you can squeeze the extra spaces before passing them to the cases. Maybe someone else from @rubocop-hq/rubocop-core has better ideas.

Reply:
You could convert the configured values into AST nodes and compare directly against `node.arguments.first`.

I played a bit with it and got as far as this:
```rb
require 'parser/current'

def on_send(node)
  return unless redundant_argument?(node)
  # ...
end

def redundant_argument?(node)
  redundant_argument = redundant_arg_for_method(node.method_name.to_s)
  return false if redundant_argument.nil?

  node.arguments.first == redundant_argument
end

def redundant_arg_for_method(method_name)
  return nil unless cop_config['RedundantArguments'].key?(method_name)

  @mem ||= {}
  @mem[method_name] ||= begin
    arg = cop_config['RedundantArguments'].fetch(method_name)

    buffer = Parser::Source::Buffer.new('(string)', 1)
    buffer.source = arg.inspect
    builder = RuboCop::AST::Builder.new
    Parser::CurrentRuby.new(builder).parse(buffer)
  end
end
```

I am not _completely_ sure that a new buffer and builder instance is needed for each configuration, so the memoization may need to be moved around a bit.